app/code/Magento/Cms/Model/ResourceModel/Page.php

```diff
@@ -265,6 +272,33 @@ protected function isValidPageIdentifier(AbstractModel $object)
         return preg_match('/^[a-z0-9][a-z0-9_\/-]+(\.[a-z0-9_-]+)?$/', $object->getData('identifier'));
     }
 
+    /**
+     *  Validate date range
+     *
+     * @param AbstractModel $object
+     * @return bool
+     */
+    protected function validateDateRange(AbstractModel $object)
```

Comment:
Please change the access level of this method to private and cover this validation with integration tests